File: backend/patchquest/orchestrator/phases.py
"""Phase definitions for the run state machine."""

from __future__ import annotations

from enum import Enum


class Phase(str, Enum):
    INTAKE = "intake"
    REPO_SCAN = "repo_scan"
    PLANNING = "planning"
    RESEARCH = "research"
    CONTEXT_BUILDING = "context_building"
    ANALYSIS = "analysis"
    PATCHING = "patching"
    STATIC_CHECKS = "static_checks"
    TESTING = "testing"
    REVIEW = "review"
    SECURITY_SCAN = "security_scan"
    FINAL_REPORT = "final_report"


class PhaseStatus(str, Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETE = "complete"
    FAILED = "failed"
    BLOCKED = "blocked"
    SKIPPED = "skipped"


PHASE_ORDER = list(Phase)

File: frontend/src/lib/phases.ts
/** Mission phase order and display labels — must match backend PHASE_ORDER. */

export const MISSION_PHASES = [
  'intake',
  'repo_scan',
  'planning',
  'research',
  'context_building',
  'analysis',
  'patching',
  'static_checks',
  'testing',
  'review',
  'security_scan',
  'final_report',
] as const

export type MissionPhase = (typeof MISSION_PHASES)[number]

export const PHASE_LABELS: Record<MissionPhase, string> = {
  intake: 'Intake',
  repo_scan: 'Cartographer',
  planning: 'Planning',
  research: 'Research',
  context_building: 'Context',
  analysis: 'Scribe',
  patching: 'Blacksmith',
  static_checks: 'Checks',
  testing: 'Trial Master',
  review: 'Inspector',
  security_scan: 'Warden',
  final_report: 'Report',
}

export const PHASE_GLYPHS: Record<MissionPhase, string> = {
  intake: '◈',
  repo_scan: '◎',
  planning: '◆',
  research: '◇',
  context_building: '▣',
  analysis: '✎',
  patching: '⚒',
  static_checks: '⌗',
  testing: '⚡',
  review: '◉',
  security_scan: '⛨',
  final_report: '▤',
}
